### PR TITLE
first attempt

### PR DESCRIPTION
**Pull request by oluoluoxenfree**
## Steps to test/reproduce
_Please explain how to best reproduce the issue and/or test the changes locally (including the pages/URLs/views/states to review)._

## Show me
_Provide screenshots/animated gifs/videos if necessary._

# REMEMBER: Attach this PR to the Trello card

### CODE REVIEW

File: plugins/postcss-cascade-layers/src/index.ts [hunk 1]
```diff
@@ -82,9 +83,11 @@ const creator: PluginCreator<undefined> = () => {
 			//  - move out styles from atRule, insert before: https://postcss.org/api/#container-insertbefore
 			//  - delete empty atRule
 			//  - give selectors the specificity they need based on layerPriority state
-			// root.walkAtRules((atRule) => {
-			// 	console.log(atRule, "third walkthrough");
-			// });
+			root.walkAtRules('layer', (atRule) => {
+				// move out styles from atRule, insert before
+
+				atRule.replaceWith(new Rule({nodes: atRule.nodes, source: atRule.source, selector: `${generateNot(layerOrder[atRule.params])}`}));
```
Comment (oluoluoxenfree) [inline plugins/postcss-cascade-layers/src/index.ts]: @romainemenke 

I know you said to use `insertbefore`, but how do i control the selector with that option? Couldn't figure it out.

Comment (sanajaved7) [reply plugins/postcss-cascade-layers/src/index.ts]: @oluoluoxenfree could we do something like this with `insertBefore` ? 
```
atRule.each((node) => {
    const transformedNode = "p:not(#\#) { max-width: 70ch; }" // use generateNot to get selector + specificity
    root.insertBefore(atRule, transformedNode);
});
atRule.remove()
```

Comment (sanajaved7) [reply plugins/postcss-cascade-layers/src/index.ts]: We could create a function to help the node selector transformation that could look something roughly like this: 
```
function transformNode(node, specificity){
  const transformedNode = `${node.selector}${generateNot(specificity)}{${node.nodes[0].prop}: ${node.nodes[0].value}}`;
  
  return transformedNode
}
```

We'll need to probably loop through the node.nodes (in my test example I just had one prop and value).

Comment (oluoluoxenfree) [reply plugins/postcss-cascade-layers/src/index.ts]: did you get any errors around it needing a source locally? 

I rewrote it to not manually create a rule, but getting different errors. 

Is what I have now along the lines of what you had in mind?

**Pull request closed without merging**